core/src/main/java/org/infinispan/stream/impl/AbstractCacheStream.java

```diff
@@ -283,24 +284,23 @@ public void close() {
             boolean localRun = ch.getMembers().contains(localAddress);
             if (localRun) {
                localValue = op.performOperation();
+
+               IntSet ourSegments;
+               if (segmentsToProcess != null) {
+                  ourSegments =  IntSets.mutableFrom(cacheTopology.getLocalReadSegments());
```

Comment:
Actually this is broken, I didn't realize that this was changed. Unfortunately we _have_ to use the primary owned segments here. Otherwise it retries too many segments - duplicating data. Unfortunately I don't see an equivalent method on the `LocalizedCacheTopology` to get the primary owned segments.

Reply:
It looks like I can just change it back to `ch.getSegmentsForOwner(localAddress)`

Reply:
I created https://issues.jboss.org/browse/ISPN-9542 to fix this and look at the remaining issues.

Reply:
Oops, I was too happy to save an allocation to notice that I had changed the results :(